apps/yggdrasil_web/web/controllers/character_controller.ex

```diff
@@ -0,0 +1,58 @@
+defmodule YggdrasilWeb.CharacterController do
+  use YggdrasilWeb.Web, :controller
+  use Guardian.Phoenix.Controller
+
+  alias Yggdrasil.Character
+
+  def index(conn, _params, user, _claims) do
+    user_id = user.id
```

Comment:
I would tend to agree with your examples.

Initially my concern was muddying up the code in the controller trying to determine what was intended but you can pattern match out the query params so may not be too bad honestly.
```
/characters # all characters I can see
#json api filter format but still only showing all for user 12
/characters?filter[user_id]=12 
```
So my only concern is that the calls `/character` and `/character?filter[user_id]=12` produce the same result for user 12 if user 12 can only see his characters, which leads to my question.

Should we possibly restrict `/characters` to a permission that can list all characters such that a user that isn't an admin for example can't see anything at all? This way we enforce that if you want to see just yours you 'must' call `/characters?filter[user_id]=12`. It feels like it may be slightly more consistent this way enforcing that idea.

Any thoughts on that? Just something I noted, it seems to lead to some ambiguity because a non admin user sees the same users either way and either call is valid for the purposes of seeing only the users characters and with and admin user only the one with the filter is valid for seeing only that users characters.

May not matter at all, just though it maybe worth mentioning.